schema/representations.md

```diff
@@ -0,0 +1,256 @@
+
+Each of these representation strategies will be explored in detail in the
+"Representation Strategy Reference" section, below.
+
+
+Representation Strategy Reference
+---------------------------------
+
+In this section, we'll go into detail on each representation strategy,
+and include examples of both schemas using it and data matching it.
+See the "Kinds and their Representations" section for an introduction to
+representation strategies.
+
+
+### struct map representation
+
+(TODO mostly interesting because defaults)
+(which maybe we should discuss more in the Kinds page?  belongs with nullable, optional, and other cardinality talk.)
+
+### struct tuple representation
+
+Tuple representation allows structs to be packed into a list representation.
+This list representation drops the struct fieldnames from the representation
+entirely and relies purely upon the order of elements to denote meaning.
+As such, it's generally a very compact way to represent data.
+
+Tuple representations should be used cautiously.  Since they contain little
+in the way of "self-describing" information, tuple representations can make for
+very fragile protocols, increase the difficulty of migrations, and make
+serialized data incomprehensible without the schema information in hand.
+
+### struct map representation example schema with defaults
+
+```ipldsch
+type TypeList struct {
+	valueType TypeTerm
+	valueNullable Bool
+} representation map {
+	field valueNullable default "false"
+}
+```
+
+#### struct tuple representation example schema
+
+```ipldsch
+```
+
+#### struct tuple representation example data
+
+```json
+```
+
+### struct stringjoin representation example schema
+
+```ipldsch
+## Fizzlebop is a pair of fields which serializes as "part1:part2" as a string.
+type Fizzlebop struct {
+	a String
+	b String
+} representation stringjoin {
+	join ":"
+}
+```
+
+### struct stringpairs representation example schema
+
+Say we're doing something awfully like the mount options in an /etc/fstab file:
+
+```ipldsch
+## MountOptions serializes as a string of comma-separated "k=v" entries, e.g. "k1=v1,k2=v2".
+## This serial representation is limited (n.b. The domain of keys must exclude "=" and values must exclude ","!),
+## but we've chosen it because other parts of our application protocol are drastically simplified by limiting this to a string.
+type MountOptions map {String:String} representation stringpairs {
+	innerDelim "="
+	entryDelim ","
+}
+```
+
+### union keyed representation example
+
+```ipldsch
+type MyKeyedUnion union {
+	| Foo "foo"
+	| Bar "bar"
+} representation keyed
+
+type Foo struct { froz Bool }
+type Bar int
+```
+
+Some data matching `MyKeyedUnion` (shown as JSON) is:
+
+```json
+{"foo": {"froz": true}}
+```
+
+This data would also match, as the other type:
+
+```json
+{"bar": 12}
+```
+
+Note how kinded unions don't introduce any kind of wrapping object at all;
```

Comment:
*keyed unions

Reply:
wait, that whole paragraph is in the wrong section :thinking: how'd that get there?

Fixed, thanks for the catch